ert_shared/ensemble_evaluator/prefect_ensemble/unix_step.py

```diff
@@ -0,0 +1,144 @@
+    storage_driver_factory,
+)
+from cloudevents.http import to_json, CloudEvent
+from ert_shared.ensemble_evaluator.entity import identifiers as ids
+
+
+class UnixStep(Task):
+    def __init__(
+        self,
+        resources,
+        outputs,
+        job_list,
+        iens,
+        cmd,
+        url,
+        step_id,
+        stage_id,
+        ee_id,
+        run_path,
+        storage_config,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._resources = resources
+        self._outputs = outputs
+        self._job_list = job_list
+        self._iens = iens
+        self._cmd = cmd
+        self._url = url
+        self._step_id = step_id
+        self._stage_id = stage_id
+        self._ee_id = ee_id
+        self._run_path = run_path
+        self._storage_config = storage_config
+
+    def get_iens(self):
+        return self._iens
+
+    def get_stage_id(self):
+        return self._stage_id
+
+    def get_step_id(self):
+        return self._step_id
+
+    def get_ee_id(self):
+        return self._ee_id
+
+    def run(self, expected_res=None):
+        if expected_res is None:
+            expected_res = []
+        else:
+            expected_res = [item for sublist in expected_res for item in sublist]
+        with Client(self._url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_FM_STEP_START,
+                    "source": f"/ert/ee/{self._ee_id}/real/{self._iens}/stage/{self._stage_id}/step/{self._step_id}",
+                    "datacontenttype": "application/json",
+                },
+            )
+
+            c.send(to_json(event).decode())
+            run_path = os.path.join(self._run_path, str(self._iens))
+
+            storage = storage_driver_factory(self._storage_config, run_path)
+            os.makedirs(run_path, exist_ok=True)
+
+            expected_res += self._resources
+            for r in expected_res:
+                storage.retrieve(r)
+
+            exec_metadata = {"iens": self._iens, "outputs": []}
+            for index, job in enumerate(self._job_list):
+                self.logger.info(f"Running command {self._cmd}  {job['name']}")
+                event = CloudEvent(
+                    {
+                        "type": ids.EVTYPE_FM_JOB_START,
+                        "source": f"/ert/ee/{self._ee_id}/real/{self._iens}/stage/{self._stage_id}/step/{self._step_id}/job/{job['id']}",
+                        "datacontenttype": "application/json",
+                    },
+                )
+                c.send(to_json(event).decode())
+                args = " ".join(job["args"])
+                shell_cmd = f"""set -e
+                #source /prog/res/komodo/stable/enable 
+                python3 {job["executable"]} {args}
+                """
+                cmd_exec = subprocess.run(
+                    shell_cmd,
```

Comment:
Think we can run the executable directly instead of of the shell script. That will require you to have a shebang and set the execution bit on you python script, but then we can run other things than Python as well. Then we can possibly get get rid of the shell=True also :slightly_smiling_face: